### PR TITLE
Change comparison ops result dtype to bool [part 1]

### PR DESCRIPTION
**Pull request by izdeby**
This is the first part of the planned changes to change the comparison operations result tensor dtype from Byte to Bool. You can see the whole list of changes (not cleaned up) [here](https://github.com/pytorch/pytorch/pull/19332). As the PR is too big for a single review im breaking it into pieces. 

**Changes in this PR:** 
1. Enable these methods for bool tensors:
- maskedSelect
- maskedSelectBool
- bitand
- cbitand
- bitor
- cbitor
- bitxor
- cbitxor
- sign
- equal
- neg

2. Add bool clause for the TH version of sign method.

Everything except the TH sign method is just a code movement with no logic change.

### CODE REVIEW

**Comment by ezyang** [comment]
Consider using `ghstack` to submit these PRs? :)

File: aten/src/TH/generic/THTensorEvenMoreMath.cpp [hunk 1]
```diff
@@ -78,6 +78,76 @@ accreal THTensor_(sumall)(THTensor *tensor)
     scalar_t, tensor, *tensor_data, sum, UNCERTAIN_TH_OMP_OVERHEAD_THRESHOLD);
   return sum;
 }
+
```
Comment (ezyang) [inline aten/src/TH/generic/THTensorEvenMoreMath.cpp]: Was the code in this block changed?

**Comment by ezyang** [comment]
The unified diff in this PR is being very unhelpful, because large chunks of code were moved around and the differ doesn't know how to play ball. I'm not sure how to solve this problem; at minimum, it would be helpful to call out where it was just code moved around, and where there were substantive changes (maybe all of this is just code movement?)

**Comment by ezyang** [comment]
I don't see any tests in this PR?

**Comment by izdeby** [comment]
@ezyang, good point on calling out what changes exactly were made. i specified which methods were touched but wasn't too clear about what exactly changed. Answering your question, yes it is only code move in this PR (except TH version of sign method, where i add check for bool scalar). Thats why i didnt add any tests as the logic didnt change.

File: aten/src/TH/generic/THTensorEvenMoreMath.cpp (hunk 1 above)
Comment (izdeby) [reply aten/src/TH/generic/THTensorEvenMoreMath.cpp]: Here and everywhere else in this PR, i just moved methods up above the `#if !defined(TH_REAL_IS_BOOL)` to enable them for bool.

**Review by ezyang — APPROVED** [review]
I assume in a later PR you will start actually exercising the bool paths.

**Pull request closed without merging**